### PR TITLE
project folder check and client presentation

### PR DESCRIPTION
**Pull request by ZimGil**
The client now imidietly ask the server to check if it's running inside an angular project,
if yes, it will find and move to the main project folder (containing `angular.json`).

### CODE REVIEW

File: server/server.ts [hunk 1]
```diff
@@ -95,3 +91,35 @@ function openBrowser(port) {
...
+  function isMainProjectFolder(response) {
+    const file = 'angular.json';
+
+    fs.access(file, fs.constants.F_OK, (err) => {
+      if (err) {
+        process.chdir('../');
+        isMainProjectFolder(response);
+      } else {
+        console.log('running inside main folder of an angular project');
+        res.send(true);
```
Comment (ortichon) [inline server/server.ts]: return true;

File: server/server.ts [hunk 2]
```diff
@@ -95,3 +91,35 @@ function openBrowser(port) {
...
+      }
+    });
+  }
+
+  function catchError(error) {
+    if (error.includes('Invalid options, "name" is required')) {
+      console.log('not running inside an angular project');
+      res.send(false);
```
Comment (ortichon) [inline server/server.ts]: return false;

File: server/server.ts [hunk 3]
```diff
@@ -95,3 +91,35 @@ function openBrowser(port) {
...
+    } else if (error.includes('This command can not be run inside of a CLI project')) {
+      console.log('running inside an angular project, looking for main folder');
+      isMainProjectFolder(res);
+    }
+  }
+
+  const testProcess = childProcess.exec('ng new', callback);
```
Comment (ortichon) [inline server/server.ts]: `callback` is redundant

File: server/server.ts [hunk 4]
```diff
@@ -95,3 +91,35 @@ function openBrowser(port) {
 
   childProcess.exec(`${start} ${url}`);
 }
+
+function isAngularProject(res) {
+  console.log('checking if running inside an agular project');
+  const callback = (err, stdout, stderr) => {};
```
Comment (ortichon) [inline server/server.ts]: remove

File: server/server.ts [hunk 5]
```diff
@@ -95,3 +91,35 @@ function openBrowser(port) {
 
   childProcess.exec(`${start} ${url}`);
 }
+
+function isAngularProject(res) {
```
Comment (ortichon) [inline server/server.ts]: this function doesn't even have to get any parameter now. we won't use `res` at all

Comment (ortichon) [reply server/server.ts]: return `bool`

File: server/server.ts [hunk 6]
```diff
@@ -95,3 +91,35 @@ function openBrowser(port) {
...
+  console.log('checking if running inside an agular project');
+  const callback = (err, stdout, stderr) => {};
+
+  function isMainProjectFolder(response) {
```
Comment (ortichon) [inline server/server.ts]: we don't need the parameter

File: server/server.ts [hunk 7]
```diff
@@ -95,3 +91,35 @@ function openBrowser(port) {
...
+    const file = 'angular.json';
+
+    fs.access(file, fs.constants.F_OK, (err) => {
+      if (err) {
+        process.chdir('../');
+        isMainProjectFolder(response);
```
Comment (ortichon) [inline server/server.ts]: same here

File: server/server.ts [hunk 8]
```diff
@@ -95,3 +91,35 @@ function openBrowser(port) {
...
+    });
+  }
+
+  function catchError(error) {
+    if (error.includes('Invalid options, "name" is required')) {
+      console.log('not running inside an angular project');
+      res.send(false);
+    } else if (error.includes('This command can not be run inside of a CLI project')) {
+      console.log('running inside an angular project, looking for main folder');
+      isMainProjectFolder(res);
```
Comment (ortichon) [inline server/server.ts]: don't pass anything

File: server/server.ts [hunk 9]
```diff
@@ -37,11 +37,7 @@ app.get('/', (req, res) => {
 });
 
 app.get('/isAngularProject', (req, res) => {
-  const file = 'angular.json';
-  
-  fs.access(file, fs.constants.F_OK, (err) => {
-    res.send(err ? false : true);
-  });
+  isAngularProject(res);
```
Comment (ortichon) [inline server/server.ts]: const isAngularProject = angularProjectChecker();

File: server/server.ts (hunk 5 above)
Comment (ortichon) [reply server/server.ts]: change the name to `angularProjectChecker`

File: src/app/app.component.ts [hunk 10]
```diff
@@ -14,6 +14,7 @@ import { CommandRequest } from './models/angular-command-request';
 export class AppComponent implements OnInit {
   title = 'Angular CLI to UI';
   isAngularProject: boolean;
+  isReadyForWork = false;
```
Comment (ortichon) [inline src/app/app.component.ts]: isLoading

**Comment by ZimGil** [comment]
This also 
resolves #43
resolves #63

**Pull request merged**